fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/values/QueriedValue.java

```diff
@@ -54,12 +61,22 @@ public QueriedValue(@Nonnull final Type resultType) {
         this.resultType = resultType;
     }
 
+    public QueriedValue(@Nonnull final Type resultType, @Nullable final Iterable<String> recordTypeNames) {
+        this.resultType = resultType;
+        this.recordTypeNames = recordTypeNames == null ? null : ImmutableList.copyOf(recordTypeNames);
+    }
+
     @Nonnull
     @Override
     public Type getResultType() {
         return resultType;
     }
 
+    @Nullable
+    public List<String> getRecordTypeNames() {
+        return recordTypeNames;
+    }
```

Comment:
this feels a bit wrong to me, why not use the `Type.Record`'s `name` member, instead it looks like we're leaking some type information from the `Value` itself.

Reply:
The type has no guaranteed name within the planner. The record type names are derived when the corresponding plans are created from their respective parts.